app/scripts/lib/ipfs-resolution/setup.js

```diff
@@ -31,26 +33,61 @@ export default function setupEnsIpfsResolver({
 
   async function webRequestDidFail(details) {
     const { tabId, url } = details;
+    console.log('details', details);
     // ignore requests that are not associated with tabs
-    // only attempt ENS resolution on mainnet
+    // only attempt resolution on mainnet
     if (tabId === -1 || getCurrentChainId() !== '0x1') {
       return;
     }
-    // parse ens name
+    // parse domain name
     const { hostname: name, pathname, search, hash: fragment } = new URL(url);
+    // if domain name contains an Unstoppable TLD, it will attempt tp resolve to IPFS
+    udTlds.forEach((tld) => {
+      if (name.toLowerCase().endsWith(`.${tld}`)) {
+        attemptResolveUns(tabId, name);
+      }
+    });
+
     const domainParts = name.split('.');
     const topLevelDomain = domainParts[domainParts.length - 1];
     // if unsupported TLD, abort
-    if (!supportedTopLevelDomains.includes(topLevelDomain)) {
+    if (!supportedEnsTopLevelDomains.includes(topLevelDomain)) {
       return;
     }
-    // otherwise attempt resolve
-    attemptResolve({ tabId, name, pathname, search, fragment });
+    // otherwise attempt resolve for ENS
+    attemptResolveEns({ tabId, name, pathname, search, fragment });
```

Comment:
@bbehrman10 i'm a noob at IPFS resolution. Will a domain that was registered via ENS be resolved through the 'attemptResolveUns' method? I'm trying to get a sense for if the order you inserted the logic is going to result in a meaningful change to the amount of traffic that MetaMask sends ENS. If its feasible that this change will result in a decrease in ENS usage from within MetaMask then we need to provide a user preference toggle for ENS vs UD resolution and re-order our checks so that it follows that preference. In that case we need to default to ENS for that preference and put in a "whats new" thing explaining that they can now use UD. In any case we can always fall back to the other if it isn't resolved via the user's preference.

So in the case of ENS preferred we attempt ENS resolution and if not then UD, if UD is preferred, reverse the paradigm.

Reply:
ENS addresses will resolve using the attemptResolveEns (it's the same function as it was before we just added the Ens identifier

Reply:
it will not impact ENS traffic at all

Reply:
my question is if a .eth address is purchased via ENS and then we attempt to resolve it using UD what will the result be?

Reply:
the way it's written, .eth will always go to ENS and the UD TLDs will all go to UD

Reply:
Hi Makoto from ENS team. Please be aware that ENS is not just `.eth`  but the users can import DNS names as ENS, such as .xyz, .io, etc.

https://medium.com/the-ethereum-name-service/announcing-support-for-xyz-on-ens-7f5bc7fe1b24
https://medium.com/the-ethereum-name-service/full-dns-namespace-integration-to-ens-now-on-mainnet-9d37270807d3

Even if UD never extends to `.eth`, it has the danger of future conflicts with the DNS imported ENS names whenever ICANN extends tlds (the last extension happened back in 2012 so the new round is expected to happen soon).